precompiles/gov/tx.go

```diff
@@ -87,3 +90,49 @@ func (p Precompile) VoteWeighted(
 
 	return method.Outputs.Pack(true)
 }
+
+// UpdateParams updates the governance parameters
+func (p Precompile) UpdateParams(
+	ctx sdk.Context,
+	origin common.Address,
+	_ *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	authority := p.govKeeper.GetAuthority()
+
+	bech32Prefix := strings.SplitN(authority, "1", 2)[0]
+	if bech32Prefix == authority {
+		return nil, fmt.Errorf("invalid bech32 address: %s", authority)
+	}
+
+	addressBz, err := sdk.GetFromBech32(authority, bech32Prefix)
+	if err != nil {
+		return nil, err
+	}
+	authorityAddress := common.BytesToAddress(addressBz)
+
+	// TODO: Not sure if the caller authority will be origin or msg.sender ?
+	// Would this require changes in the SDK so that transactions from the gov handler go through the Precompile ?
+	if origin != authorityAddress {
```

Comment:
I wouldnt merge until we solve this TODO

Reply:
Yeah this will require a bit more thought as it will require re-wiring the gov handler, which means a wrapper or a modification to our Cosmos SDK fork. I'll let you guys decide that but you can always just remove this tx from the precompile handler, merge it and figure out what the approach is later.